consensus/polybft/state_sync_manager.go

```diff
@@ -60,14 +62,14 @@ var _ StateSyncManager = (*stateSyncManager)(nil)
 type stateSyncManager struct {
 	logger hclog.Logger
 	state  *State
-
 	config *stateSyncConfig
 
 	// per epoch fields
-	lock         sync.Mutex
-	commitment   *Commitment
-	epoch        uint64
-	validatorSet *validatorSet
+	lock               sync.RWMutex
+	pendingCommitments []*Commitment
```

Comment:
What is the lifecycle of this field? It seems to me that if the process gets shutdown, the system would not recover pending commitments?

Reply:
We do not save this anywhere but in memory. When we get back up, we will build new pending commitments on restart epoch (if we get back up, and the chain is in a newer epoch), or we will build new pending commitments on every state sync event that the event tracker gets. In both cases, we might not build the smallest commitments that the other validators have, depending on the state sync events we have and get when event tracker pulls everything we missed.

Reply:
Ok, from what I see on the RFC too. If the node gets shutdown it "resets" the `pendingCommitments` in two ways: either waits for the next log or for the next epoch. Am I right? If that is correct, it is good for me.

Reply:
That is right.